### DISCUSSION TITLE
Cake in offline environment.

### DISCUSSION THREAD

**Discussion by Eldadc** [cake-build/cake#3035] [Q&A]
Hi Team,

I am setting up cake script in Jenkins CI system in offline environment ,(i.e. : no connection to the outside internet )
I was able to do so with editing few steps in the build.ps1 script. I am still have some issue when the script does not check if the build need to restore nugets , although it needs. ( SkipToolPackageRestore,)
I was wondering Is there a manual on how to do it as best practice.

Thanks

**Comment by gep13** [comment]
@eldadc this needs to be written up into an article, but i recorded a video about this here:

https://youtu.be/wkPN9CZCZFM

**Comment by Eldadc** [comment]
THANKS .

**Comment by Eldadc** [comment]
Thanks on the detailed session.
I have 2 issues that were not covered.
What are the changed in the script if I put the package.config and nuget in UNC folder in the network. ?
Today my package.config is part of the project i notice that the SkipToolPackageRestore is set always skipping nuget restore.

**Comment by gep13** [comment]
@Eldadc you should not need to make any changes to the packages.config file.  For using a UNC path, the required changes should be similar to those that were made in the video.  In the video I had assumed that the source would need to be changed to be an internal NuGet Server, for example Nexus.  If using a UNC path, you should be able to switch to using that, rather than the http address of the NuGet server.

I am not sure what you mean by this part:

> Today my package.config is part of the project i notice that the SkipToolPackageRestore is set always skipping nuget restore.

Can you elaborate?

If you wanted to, I would be happy to talk about this further: https://officehours.io/people/gep13

**Comment by Eldadc** [comment]
Hi,

Thanks on the detailed answer and follow up.
UNC - got it ,  Do I need any adjustment with the PS syntax ?

I debug my build.ps1 and I noticed that SkipToolPackageRestore will skip the restore package commands although there are no packages.  I store my tools\package.config file in my source control.
Do I need to edit also the SkipToolPackageRestore switch ? , Can I remove the "if" check statement and always restore nuget packages.

Thanks In Advanced,

**Comment by gep13** [comment]
Have a look here:

https://blog.nuget.org/20150922/Accelerate-Package-Source.html

For further information.

No, you shouldn't need to edit any of that.  In order for your build to work, you will still need to restore things, it is just that you are restoring from an internal location, rather than from the internet.
